### PR TITLE
fix: Make testAsyncStacktraces pass again 

### PR DESCRIPTION
**Pull request by kevinrenskers**
See https://github.com/getsentry/sentry-cocoa/pull/2142#discussion_r969392752

#skip-changelog

### CODE REVIEW

File: Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift [hunk 1]
```diff
@@ -103,14 +100,13 @@ class SentryStacktraceBuilderTests: XCTestCase {
         let filteredFrames = actual.frames.filter { frame in
             return frame.function?.contains("testAsyncStacktraces") ?? false ||
             frame.function?.contains("asyncFrame1") ?? false ||
-            frame.function?.contains("asyncFrame2") ?? false ||
-            frame.function?.contains("asyncAssertion") ?? false
```
Comment (brustolin) [inline Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift:107]: But why we used to expected to have an asyncAssertion and now we dont?

Comment (kevinrenskers) [reply Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift:107]: I don't know, but I do know that `asyncAssertion` is never part of the stack traces, and removing this check is better than the test being completely disabled.

Comment (brustolin) [reply Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift:107]: @philipphofmann any idea?

Comment (philipphofmann) [reply Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift:107]: I have no clue. The feature is anyways experimental, and we need to have another look at it. I'm fine with this change, as long as the test runs again. Thanks, @kevinrenskers

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**